rest-api/offline/public_metrics_export.py

```diff
@@ -205,3 +215,25 @@ def export():
               'count': row.items()[1][1],
           })
     return out
+
+  @staticmethod
+  def _save(metric_set_id, metrics):
+    ms = MetricSet(
+        metricSetId=metric_set_id,
+        metricSetType=MetricSetType.PUBLIC_PARTICIPANT_AGGREGATIONS,
+        lastModified=clock.CLOCK.now()
+    )
+    with database_factory.get_generic_database().session() as session:
+      MetricSetDao().upsert_with_session(session, ms)
+      agg_dao = AggregateMetricsDao()
+      agg_dao.delete_all_for_metric_set_with_session(session, metric_set_id)
```

Comment:
I guess if we had a guarantee that all keys in previous metrics were present in new metrics, we wouldn't technically need this delete (could use an upsert below instead). But I agree this is probably safer.

Reply:
Yes, this could happen if we happened to have an aggregation with maybe ~1 person, then that person withdrew or we updated some information in the RDR which moved them out of that group.